File: LeetCode/35-search-insert-position.py
###############################################################################################
# easy
###########
# Time Complexity: O(logn)
# Space Complexity: O(1)
###############################################################################################
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1
        while l < r:
            m = (l+r)>>1
            if nums[m] >= target:
                r = m
            else:
                l = m+1
        return l if nums[l] >= target else l+1

File: LeetCode/35-search-insert-position.cpp
///////////////////////////////////////////////////////////////////////////////////////////////
// easy
////////////////
// Time Complexity: O(logn)
// Space Complexity: O(1)
///////////////////////////////////////////////////////////////////////////////////////////////
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int i=0, j=nums.size()-1, mid;
        while (i < j) {
            mid = (i+j)>>1;
            if (nums[mid] < target) i = mid+1;
            else j = mid;
        }
        return nums[i]>=target?i:i+1;
    }
};